curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -812,31 +812,28 @@ private void performBackgroundOperation(OperationAndData<?> operationAndData)
                 queueOperation(operationAndData);
             }
         }
-        catch ( Throwable e )
+        catch ( CuratorConnectionLossException e )
```

Comment:
Similar to above. If Throwable is not caught, the exception will be lost. By catching Throwable, the exception will get logged.